pywbem/_cim_operations.py

```diff
@@ -2148,37 +2254,171 @@ def _iparam_classname(classname, arg_name):
             classname.namespace = None
         elif isinstance(classname, six.string_types):
             classname = CIMClassName(classname)
-        elif classname is None:
+        elif not required and classname is None:
             pass
         else:
             raise TypeError(
-                _format("The {0!A} argument of the WBEMConnection operation "
-                        "has invalid type {1} (must be None, a string, or a "
+                _format("The {0!A} parameter of the WBEMConnection operation "
+                        "has invalid type {1} (must be {2}a string, or a "
                         "CIMClassName)",
-                        arg_name, type(classname)))
+                        arg_name, type(classname),
+                        "" if required else "None, "))
         return classname
 
     @staticmethod
-    def _iparam_instancename(instancename):
+    def _iparam_instancename(instancename, arg_name, required=False):
         """
-        Convert an instance name specified in an operation method into a CIM
-        object that can be passed to imethodcall().
+        Convert an instance name specified in an operation method into a
+        CIMInstanceName object that can be passed to imethodcall().
+
+        Returns:
+          * CIMInstanceName without host and namespace, if instancename is
+            CIMInstanceName
+          * None, if instancename is None (only if not required)
+
+        Raises:
+          TypeError - instancename has an invalid type
         """
 
         if isinstance(instancename, CIMInstanceName):
             instancename = instancename.copy()
             instancename.host = None
             instancename.namespace = None
-        elif instancename is None:
+        elif not required and instancename is None:
             pass
         else:
             raise TypeError(
-                _format("The 'InstanceName' argument of the WBEMConnection "
-                        "operation has invalid type {0} (must be None, a "
+                _format("The 'InstanceName' parameter of the WBEMConnection "
+                        "operation has invalid type {0} (must be {1}a "
                         "string, or a CIMInstanceName)",
-                        type(instancename)))
+                        type(instancename),
+                        "" if required else "None, "))
         return instancename
 
+    @staticmethod
+    def _iparam_class(klass, arg_name):
+        """
+        Convert a CIMClass object specified in an operation method such that
+        it can be passed to imethodcall().
+
+        Returns:
+          * CIMClass copy without path
+          * None
+
+        Raises:
+          TypeError - klass has an invalid type
+        """
+
+        if isinstance(klass, CIMClass):
+            klass = klass.copy()
```

Comment:
Not sure but asking.  Should this be deep copy?

Reply:
NOISSUE - No.
We are only modifying the path, so all the rest of the object remains unchanged. When used for CIM-XML, it will not be stored longer term (and will not further be modified), and when used for the mock repo, it will be deep-copied upon adding it to the repo.